src/main/scala/it/pagopa/interop/apigateway/api/impl/AuthApiServiceImpl.scala

```diff
@@ -73,14 +74,15 @@ final case class AuthApiServiceImpl(
     } yield (m2mToken, checker)
 
     val result: Future[ClientCredentialsResponse] = for {
-      (m2mToken, checker)       <- tokenAndChecker.toFuture
+      (m2mToken, checker) <- tokenAndChecker.toFuture
+      m2mContexts = Seq(CORRELATION_ID_HEADER -> UUID.randomUUID().toString, BEARER -> m2mToken)
```

Comment:
At this level, for this specific authorization endpoint, we do not expect a correlation id from the caller, so we're programmatically generating a contexts Seq with a random Correlation id.